### PR TITLE
WIP Improve docstrings / type hints in CellComplex

### PR DESCRIPTION
**Pull request by josefhoppe**
I've started improving docstrings in CellComplex (#95) and improved type hints (#102) along the way.

### CODE REVIEW

**Review by ninamiolane — APPROVED** [review]
Excellent, thanks!

File: toponetx/classes/cell_complex.py [hunk 1]
```diff
@@ -11,12 +11,14 @@
 from collections import defaultdict
 from collections.abc import Hashable, Iterable
 from itertools import pairwise, zip_longest
+from typing import Any, Dict, Optional, Set, Tuple
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:14]: TopoNetX is Python 3.9+, so you can use the native `dict`, `set` and `tuple` classes with subscripting. The dedicated classes in `typing` have been deprecated.

File: toponetx/classes/cell_complex.py [hunk 2]
```diff
@@ -270,7 +272,7 @@ def __getitem__(self, node):
         """
         return self.neighbors(node)
 
-    def _insert_cell(self, cell, **attr):
+    def _insert_cell(self, cell: tuple | list | Cell, **attr):
```
Comment (ffl096) [inline toponetx/classes/cell_complex.py:275]: TopoNetX is Python 3.9+, so you cannot use union types using the pipe character (Python 3.10+ only). Use `typing.Union`.

File: toponetx/classes/cell_complex.py (hunk 1 above)
Comment (josefhoppe) [reply toponetx/classes/cell_complex.py:14]: Oh, my bad. I thought those were still up-to-date, thank you.

**Pull request merged**